test/model/autocomplete_test.dart

```diff
@@ -178,8 +184,106 @@ void main() {
     doTest('~@_Rodion Romanovich Raskolniko^', silentMention('Rodion Romanovich Raskolniko'));
     doTest('~@Родион Романович Раскольников^', mention('Родион Романович Раскольников'));
     doTest('~@_Родион Романович Раскольнико^', silentMention('Родион Романович Раскольнико'));
-    doTest('If @chris is around, please ask him.^', null); // @ sign is too far away from cursor
-    doTest('If @_chris is around, please ask him.^', null); // @ sign is too far away from cursor
+
+    // "#" sign can be (3 + 2 * maxChannelName) utf-16 code units
+    // away to the left of cursor.
+    doTest('If ~@chris^ is around, please ask him.', mention('chris'), maxChannelName: 10);
+    doTest('If ~@_chris is^ around, please ask him.', silentMention('chris is'), maxChannelName: 10);
+    doTest('If @chris is around, please ask him.^', null, maxChannelName: 10);
+    doTest('If @_chris is around, please ask him.^', null, maxChannelName: 10);
+
+    // #channel link.
+
+    doTest('^#', null);
+    doTest('^#abc', null);
+    doTest('#abc', null); // (no cursor)
+
+    doTest('~#^', channelLink(''));
+    doTest('~##^', channelLink('#'));
+    doTest('~#abc^', channelLink('abc'));
+    doTest('~#abc ^', channelLink('abc '));
+    doTest('~#abc def^', channelLink('abc def'));
+
+    // Accept space before channel link syntax.
+    doTest(' ~#abc^', channelLink('abc'));
+    doTest('xyz ~#abc^', channelLink('abc'));
+
+    // Accept punctuations before channel link syntax.
```

Comment:
nit:
```suggestion
    // Accept punctuation before channel link syntax.
```

"Punctuation" is a [mass noun](https://en.wikipedia.org/wiki/Mass_noun), like "water".